cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -46,6 +46,8 @@ type ScaleSet struct {
 	mutex       sync.Mutex
 	lastRefresh time.Time
 	curSize     int64
+	// virtualMachines holds a list of vmss instances (instanceID -> resourceID).
+	virtualMachines map[string]string
```

Comment:
Isn't this generic map enough that we need to move this to azure manager and use this in non-vmss cases ?

Reply:
No, this is only for vmss.